src/ast/stmt/SwitchStmt.php

```diff
@@ -64,4 +65,22 @@ public function injectScope(&$parent_scope)
             $case->injectScope($parent_scope);
         }
     }
+
+    public function runTypeChecker()
+    {
+        $value_type = $this->value->getType();
+        foreach ($this->cases as $case) {
+            if (!$case->is_else) {
+                $case_type = $case->value->getType();
+                if ($case_type->code !== $value_type->code) {
+                    // TODO: how to get the case expression value? 
```

Comment:
Simply use `$case->value`. It contains the AST node for the expression. It may be found [here](https://github.com/quack/quack/blob/master/src/ast/stmt/CaseStmt.php). You  can also write a `runTypeChecker` implementation for `CaseStmt`, that calls it for each node in the body :smile: